claim-check/code-samples/sample-2/client-consumer/EventHubsConsumer.cs

```diff
@@ -0,0 +1,126 @@
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.EventHubs;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Newtonsoft.Json.Linq;
+
+namespace ClientConsumer
+{
+    class PartitionReceiver : IPartitionReceiveHandler
+    {
+        readonly CloudBlobClient _cloudBlobClient = null;
+        readonly string _downloadDestination = string.Empty;
+        int _maxBatchSize = 1;
+        public int MaxBatchSize { get => _maxBatchSize; set => _maxBatchSize = value; }
+
+        public PartitionReceiver(CloudStorageAccount storageAccount, string downloadDestination)
+        {
+            _cloudBlobClient = storageAccount.CreateCloudBlobClient();
+            _downloadDestination = downloadDestination;
+        }
+
+        public Task ProcessErrorAsync(Exception error)
+        {
+            Console.WriteLine(error.Message);
+            return Task.FromException(error);
+        }
+
+        public Task ProcessEventsAsync(IEnumerable<EventData> events)
+        {
+            foreach(var e in events)
+            {
+                string body = Encoding.UTF8.GetString(e.Body);
+
+                var jsonMessage = JArray.Parse(body).First;
+
+                Uri uploadedUri = new Uri(jsonMessage["data"]["url"].ToString());
+                Console.WriteLine("Blob available at: {0}", uploadedUri);
+                var cloudBlob = _cloudBlobClient.GetBlobReferenceFromServer(uploadedUri);
+
+                string uploadedFile = Path.GetFileName(jsonMessage["data"]["url"].ToString());
+                string destinationFile = Path.Combine(_downloadDestination, Path.GetFileName(uploadedFile));
+                Console.WriteLine("Downloading to {0}...", destinationFile);
+                cloudBlob.DownloadToFile(destinationFile, FileMode.Create);
+                Console.WriteLine("Done.");
+                Console.WriteLine();
+            }
+
+            return Task.CompletedTask;
```

Comment:
This is not needed when the above changes are made.

Reply:
Removed

Reply:
The return keyword is no longer needed since you have made the method async, so please remove it.

Reply:
Done